### PR TITLE
Flows for Q&A

### PR DESCRIPTION
**Pull request by smcnally**

### CODE REVIEW

**Comment by smcnally** [comment]
Thanks, again, for making chat-bubble available. 
Make or request changes as you see fit.

File: examples/6-advanced-flows.html [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+            "[Answering 'Yes' here will open a new window with a page from another site]", // call a function to an external resource / application outside of chat-bubble's purview
+        ],
+        "reply": [
+            {
+                "question": "Not interested, thanks. [Escape Valve hit]",
+                "answer": "end"
+            },
+            {
+                "question": "Yes",
+                "answer": "externalResourceFunction"
```
Comment (dmitrizzle) [inline examples/6-advanced-flows.html]: I think this function should also give user an option to continue chat. Otherwise the window will pop up and there will be no way to continue from here on.

File: examples/6-advanced-flows.html [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+        ]
+    },
+    // another conversation object that can be queued from within
+    // any other conversation object, including itself
+} // end conversation object
+)
+
+    // this function is called when user asks for banana
+    externalResourceFunction = function () {
+        window.open("https://meanbusiness.com/gamepath/coralproject/coral-swiper-180306/","_blank",
```
Comment (dmitrizzle) [inline examples/6-advanced-flows.html]: What is this website?

File: examples/6-advanced-flows.html [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+      // pass your JSON/JavaScript object to `.talk()` function where
+// you define how the conversation between the bot and user will go
+{
+    // "ice" (as in "breaking the ice") is a required conversation object
+    // that maps the first thing the bot will say to the user
+        ice: {
+        // "says" defines an array of sequential bubbles
+        // that the bot will produce
+        says: [
+            "Hello",
```
Comment (dmitrizzle) [inline examples/6-advanced-flows.html]: The way the users will see this example is by clicking the link without any of the comments. I think it's important to let them know that this is an example chat flow that they could explore some advanced features from.

File: examples/6-advanced-flows.html [hunk 4]
```diff
@@ -0,0 +1,213 @@
...
+        ],
+        // "reply" is an array of possible options the user can pick from
+        // as a reply
+      reply: [
+            {
+        question: "Sure", // label for the reply option
+    answer: "chapter-one" // key for the next conversation object
+            },
+            {
+        question: "No. Thanks [Escape Valve hit]", // label for the reply option
```
Comment (dmitrizzle) [inline examples/6-advanced-flows.html]: Not sure if the information in the square brackets is helpful in this chat. Perhaps we can explain in plain English what the bot is doing?

File: examples/6-advanced-flows.html [hunk 5]
```diff
@@ -0,0 +1,213 @@
...
+            }
+        ]
+    },
+    "chapter-one": {
+        "says": [
+            "Excellent -- Thank you.",
+            "[Questions | Info | Requests for feedback]",
+        ],
+        "reply": [
+            {
```
Comment (dmitrizzle) [inline examples/6-advanced-flows.html]: This set (and the following few) of answers is a little confusing. I understand this is an example, but perhaps we could make up some imaginary questions and answers to match closer to an actual use-case?

**Review by dmitrizzle — CHANGES_REQUESTED** [review]
Thank you so much for this! I've added some comments in the code which I think would be best addressed before release. In general, I think it would be good to format this chat as an actual example that the users may enjoy following and participating in. A bonus would be giving the a link to the code at the end so that they could follow the comments and, perhaps, attempt to build a similar bot.

🎉

**Comment by smcnally** [comment]
All makes sense. Acknowledging here. Will update.

File: examples/6-advanced-flows.html (hunk 5 above)
Comment (smcnally) [reply examples/6-advanced-flows.html]: I can keep the narrative to Bananas and Ice Cream for continuity with your other examples. 

The chat is based on [an example use case](https://meanbusiness.com/gamepath/chat-bubble/examples/Coral-intro-chat.html) I did for the [Coral Project](https://www.coralproject.net/) to represent personality types in News communities.

File: examples/6-advanced-flows.html (hunk 4 above)
Comment (smcnally) [reply examples/6-advanced-flows.html]: Understood. As you note above, the majority of people will experience the example front-end first, rather than via code comments. I'll update accordingly.

File: examples/6-advanced-flows.html (hunk 3 above)
Comment (smcnally) [reply examples/6-advanced-flows.html]: I'll update accordingly.

File: examples/6-advanced-flows.html (hunk 2 above)
Comment (smcnally) [reply examples/6-advanced-flows.html]: [As below](https://github.com/dmitrizzle/chat-bubble/pull/53#discussion_r179615446), this is another example use case I did for the Coral Project to represent personality types in News communities. Will update this link.

File: examples/6-advanced-flows.html (hunk 1 above)
Comment (smcnally) [reply examples/6-advanced-flows.html]: Will add that.

File: examples/6-advanced-flows.html (hunk 5 above)
Comment (dmitrizzle) [reply examples/6-advanced-flows.html]: This is awesome, love the project! How about using the bananas & ice cream for now; but would you like to add a link to your project to the main Readme.md?

File: examples/6-advanced-flows.html (hunk 1 above)
Comment (smcnally) [reply examples/6-advanced-flows.html]: I've addressed all but this one and can use guidance re getting start over / ice to work within the script.

**Comment by smcnally** [comment]
I've  kept the narrative to Bananas and Ice Cream, added the flow comments on the front end, removed the square brackets. All but [adding Start Over](https://github.com/dmitrizzle/chat-bubble/pull/53#discussion_r179360302) to the script should be handled. Please say the word if you'd like more changes.

**Comment by dmitrizzle** [comment]
Hi @smcnally thank you for the changes!
I've edited your example a bit for clarity and have fixed the external function so that it does not interrupt the chat. If this looks good to you, I'll add more detailed comments in to the code and merge.

**Review by dmitrizzle — APPROVED** [review]

**Comment by dmitrizzle** [comment]
Note to self: should add dimensions to GIFs to ensure that they do not jump when loaded.

**Review by smcnally — COMMENTED** [review]
Looks good. Thank you.
